FuelLabs/fuel-indexer
# fix (fuel-indexer,fuel-indexer-types): Re-integrate block producer

Closes #334.

## Changelog
- Re-integrates block producer into `BlockData`

## Notes
Since #421 is blocked by the proverbial dependency hell, I elected to do this in its own PR to move things forward.

## Testing Plan
1. Run test components.
2. Start indexer service with block explorer example.
3. Generate blocks for test data.
4. Confirm that `producer` is present when adding values to database.
